_news/2019-10-07-3.1-release-notes/en.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Dodona 3.1"
+description: "Release notes Dodona 3.1"
+slug: "dodona-3.1"
+date: "2019-10-07"
+lang: en
+---
+
+The release of Dodona 3.1 makes Dodona more secure and faster.
+
+## Security
+
+Until now, exercises could change the entire layout of Dodona and execute _scripts_. These _scripts_ were able to do anything the user looking at the page could do on Dodona, e.g. submit solutions. For teachers this meant that these _scripts_ could be used to extract data about students. To be clear, this has never happened. However, since an ever increasing number of people can add exercises to Dodona, it became time to fix this.
```

Comment:
```suggestion
Until now, exercises could change the entire layout of Dodona and execute _scripts_. These _scripts_ were able to do anything the user looking at the page could do on Dodona (e.g., submit solutions) which is a potential security risk. While we fully trust our exercise creators and don't have any evidence that this feature was misused, it is better to prevent any future problems.
```